spec/ddtrace/contrib/ethon/ethon_patch_spec.rb

```diff
@@ -0,0 +1,356 @@
+require 'spec_helper'
+require 'ddtrace/contrib/analytics_examples'
+
+require 'ddtrace'
+require 'ddtrace/contrib/ethon/easy_patch'
+require 'ddtrace/contrib/ethon/multi_patch'
+require 'typhoeus'
+require 'stringio'
+require 'webrick'
+
+RSpec.describe Datadog::Contrib::Ethon do
```

Comment:
There's a lot of good tests in here, but I think I'd like to see it broken out into smaller specs if possible. Generally speaking I like to see separate unit tests from the integration tests, then any other test subjects into their own specs. In this case something like:

 - `integration_spec.rb`: Integration tests for Ethon (mostly what you have here)
 - `typhoeus_integration_spec.rb`: Integration tests that use Typhoeus (what you have at the bottom)
 - `easy_spec.rb`: Unit tests for the Easy patch
 - `multi_spec.rb` Unit tests for the Multi patch

In the unit tests, I would try to mock/stub to avoid any external web calls (which tend to flake out in CI.) Using WEBrick in the integration tests is cool with me, but we should probably add `before { skip unless ENV['TEST_DATADOG_INTEGRATION'] }` at the top of them so they don't run in non-integration safe environments.